### PR TITLE
Added support for comments

### PR DESCRIPTION
**Pull request by eblanchette**
Please have a look at my changes for comment support.  You will need to create a comment "Hello World!" on your test page. I've never used Kotlin before, so please let me know if there are any issues.

image

image

### CODE REVIEW

**Comment by seratch** [comment]
This pull request may resolve #64

File: core/src/test/kotlin/OAuthAppExample.kt [hunk 1]
```diff
@@ -33,7 +33,7 @@ class StateManager {
   }
 }
 
-fun urlEncode(v: String?): String = URLEncoder.encode(v, Charsets.UTF_8)
+fun urlEncode(v: String?): String = URLEncoder.encode(v, "UTF-8")
```
Comment (seratch) [inline core/src/test/kotlin/OAuthAppExample.kt:36]: Why did you change this?

File: core/src/test/kotlin/integration_tests/CommentsTest.kt [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        comments.addAll(children.results)
+        cursor = children.nextCursor
+      }
+
+      var found = false
+      val itr = comments.listIterator()    // or, use `iterator()`
+      while (itr.hasNext()) {
+          var comment = itr.next()
+          for(richText in comment.richText) {
+              println(richText.text)
```
Comment (seratch) [inline core/src/test/kotlin/integration_tests/CommentsTest.kt:46]: please remove println

File: core/src/test/kotlin/integration_tests/CommentsTest.kt [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
```
Comment (seratch) [inline core/src/test/kotlin/integration_tests/CommentsTest.kt:94]: same as above

**Review by seratch — CHANGES_REQUESTED** [review]
@eblanchette Thanks for sending this PR! 

I am sorry that I am a bit busy with other personal tasks on the weekends. So, I may not be able to make efforts on this project within a few weeks. When I can use my time for this repo, I will review your changes here in detail. That being said, your changes look great to me at a glance. 

I left a few comments for now. Also, can you apply the code formatted? https://github.com/seratch/notion-sdk-jvm/blob/main/scripts/format.sh

File: core/src/test/kotlin/integration_tests/CommentsTest.kt (hunk 3 above)
Comment (eblanchette) [reply core/src/test/kotlin/integration_tests/CommentsTest.kt:94]: Sorry, will be removed.

File: core/src/test/kotlin/integration_tests/CommentsTest.kt (hunk 2 above)
Comment (eblanchette) [reply core/src/test/kotlin/integration_tests/CommentsTest.kt:46]: Sorry, will be removed.

File: core/src/test/kotlin/OAuthAppExample.kt (hunk 1 above)
Comment (eblanchette) [reply core/src/test/kotlin/OAuthAppExample.kt:36]: When I was initially compiling the project, I was using JDK 8.  I was getting the following error:

```
[ERROR] /.../eblanchette/notion-sdk-jvm/core/src/test/kotlin/OAuthAppExample.kt:[36,58] Type mismatch: inferred type is Charset but String! was expected
```

I have since updated to JDK 11 and the error goes away.  I will revert my changes.

**Comment by eblanchette** [comment]
Please review my changes.

**Comment by seratch** [comment]
The change should be fine but just in case, I will look into it on my end this weekend. Thanks for being patient here!

**Comment by seratch** [comment]
@eblanchette Thanks a lot for taking the time to make this change! I've merged your change https://github.com/seratch/notion-sdk-jvm/commit/b5513556eef493de00d341fe8f517bf3685f7d34
